themes/default/content/registry/packages/kubernetes/how-to-guides/choosing-the-right-helm-resource-for-your-use-case.md

```diff
@@ -0,0 +1,101 @@
+
+### Benefits
+
+1. Visibility into all resources encapsulated by the Chart in Pulumi's state, allowing users the ability to directly query properties of individual resources.
+2. The above also allows Pulumi's Policy-as-Code framework - [`CrossGuard`]({{< relref "/docs/guides/crossguard/" >}}) to enforce policies on all resources installed by Helm charts
+3. Ability to leverage transformations to programmatically manipulate resources installed by Helm charts in any of the Pulumi supported programming languages
+4. Detailed previews and diffs rendered in the Pulumi CLI and Console for each Kubernetes resource resulting from Helm Chart config changes
+
+We have seen significant adoption of these resources over the years. However, these resources also come with some limitations which have affected some of our Kubernetes users:
+
+### Limitations
+
+1. No support for [Helm Chart Hooks](https://helm.sh/docs/topics/charts_hooks/) - i.e. equivalent of running `helm install` with the `--no-hooks` option
+2. No ability to import existing Helm releases into Pulumi state
+3. No ability to interoperate using the Helm CLI on resources installed by Pulumi
+
+## Helm Release Resource
+
+The Pulumi Kubernetes provider uses an embedded version of the Helm SDK to natively manage [`Helm Releases`](https://helm.sh/docs/glossary/#release) on the target Kubernetes cluster. The addresses many of the main limitations with the `Chart` resources mentioned above:
+
+### Benefits
+
+1. Since we use Helm's native support for downloading, processing and installing charts, all the major features of Helm Charts such as hooks can be readily supported
+2. Existing Helm releases installed via the Helm CLI can be imported into Pulumi state as of [v3.12.1](https://github.com/pulumi/pulumi-kubernetes/releases/tag/v3.12.1) of the Pulumi Kubernetes SDK
+3. Releases installed via Pulumi are serialized by the chosen Helm driver in the cluster and can be queried by the Helm CLI.
+
+However, in return for these improvements, Helm Release resource incurs a few inherrent limitations:
+
+### Limitations
+
+1. The ability to support transformations is limited to that offered by the `Helm` CLI (running `helm install` with `--post-renderer postrenderer`)
+2. Pulumi doesn't have direct access to the underlying Kubernetes resources installed by `Helm`, as a result, `CrossGuard` policies can't be enforced on Helm Release resources.
+3. Pulumi doesn't render fine-grained diffs in previews scoped to affected Kubernetes resources when Helm configuration changes.
```

Comment:
```suggestion
3. Fine-grained preview support is limited, since the Helm client is responsible for managing the underlying Kubernetes resources.
```